test/yb-ctl-test.sh

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+export YB_PG_FALLBACK_SYSTEM_USER_NAME=$USER
+export YB_DISABLE_CALLHOME=1
+
+# This will be auto-detected the first time yb-ctl auto-downloads and installs YugaByte DB.
+installation_dir=""
+
+log() {
+  echo >&2 "[$( date +%Y-%m-%dT%H:%M:%S )] $*"
+}
+
+detect_installation_dir() {
+  if [[ -z $installation_dir ]]; then
+    installation_dir=$( ls -td "$HOME/yugabyte-db/yugabyte-"* | head -1 )
+    log "YugaByte DB has been automatically installed into directory: $installation_dir"
+  fi
+}
+
+verify_ysqlsh() {
+  local node_to_connect=${1:-}
+  log "Creating a YSQL table and inserting a bit of data"
+  local ysqlsh_opts=""
+  if [[ -n $node_to_connect ]]; then
+    ysqlsh_opts="-h 127.0.0.$node_to_connect"
+  fi
+  "$installation_dir"/bin/ysqlsh $ysqlsh_opts <<-EOF
+create table mytable (k int primary key, v text);
+insert into mytable (k, v) values (10, 'sometextvalue');
+insert into mytable (k, v) values (20, 'someothertextvalue');
+EOF
+  log "Running a simple select from our YSQL table"
+  echo "select * from mytable where k = 10; drop table mytable;" | \
+    "$installation_dir"/bin/ysqlsh $ysqlsh_opts | \
+    grep "sometextvalue"
+}
+
+start_cluster_run_tests() {
+  if [[ $# -ne 1 ]]; then
+    fatal "One arg expected: root directory to run in"
+  fi
+  local root_dir=$1
+  ( set -x; "$root_dir"/yb-ctl "${yb_ctl_args[@]}" start )
+  verify_ysqlsh
+  ( set -x;  "$root_dir"/yb-ctl "${yb_ctl_args[@]}" add_node )
+  verify_ysqlsh
+  ( set -x; "$root_dir"/yb-ctl "${yb_ctl_args[@]}" stop_node 1 )
+  # TODO: investigate why we can't connect and send YSQL commands to the second node here.
+  # Perhaps the master gets shut down with the first node?
```

Comment:
Interesting. Tried this manually and I'm able able to connect to the second node via ysqlsh. stop_node only stops the tserver and not master (if you don't specify --master).

Reply:
if I understand the code above correctly (check other comment), we could fix this by either stopping node 2 instead (so we keep the verify default behavior), or we explicitly pass 2 as the arg to verify, after this stop...